tests/services/test_ingestion.py

```diff
@@ -493,3 +496,39 @@ def test_poll_upload_merge_status(app, db, test_user, monkeypatch):
         assert (
             "status_details" in res.json and res.json["status_details"] == test_details
         )
+
+
+def test_extra_metadata(
+        app_no_auth, monkeypatch
+):
+    """Ensure the extra assay metadata endpoint follows the expected execution flow"""
+    from cidc_api.services.ingestion import AssayUploads as _AssayUploads
+
+    merge_extra_metadata = MagicMock()
+    monkeypatch.setattr(_AssayUploads, "merge_extra_metadata", merge_extra_metadata)
+
+    client = app_no_auth.test_client()
+    res = client.post('/ingestion/extra-assay-metadata')
+    assert res.status_code == 400
+    assert 'Expected form' in res.json['_error']['message']
+
+    res = client.post('/ingestion/extra-assay-metadata', data={'foo': 'bar'})
+    assert res.status_code == 400
+    assert 'job_id' in res.json['_error']['message']
+
+    res = client.post('/ingestion/extra-assay-metadata', data={'job_id': 123})
+    assert res.status_code == 400
+    assert 'files' in res.json['_error']['message']
+
+    form = {
+        'job_id': 123,
+        'uuid-1': (io.BytesIO(b'fake file 1'), 'fname1'),
+        'uuid-2': (io.BytesIO(b'fake file 2'), 'fname2')
+    }
+
+    res = client.post('/ingestion/extra-assay-metadata', data=form)
+
+    assert res.status_code == 200
+    merge_extra_metadata.assert_called()
```

Comment:
please add assertions on data being actually updated/saved.

Reply:
also check validation errors being handled and maybe converted to bad request errors with human comprehensible message.

Reply:
BadRequest errors being raised [here](https://github.com/CIMAC-CIDC/cidc-api-gae/blob/e4153aca6619be85e01c95fcd48d1fbb8a7f6ab4/cidc_api/services/ingestion.py#L524)

Reply:
the prism call result updates the existing patch (referred [here](https://github.com/CIMAC-CIDC/cidc-schemas/blob/f8dfa6c63076a95d6f8e155f89998431981d4caa/cidc_schemas/prism.py#L786))